### ISSUE TITLE
.animate() does not return Promise

### ISSUE THREAD

**Issue by hradhak** [ractivejs/ractive#2776]
## Description:
When calling animate() undefined is returned instead of a Promise.   #2382 was closed so I'm not sure if this is a regression or a completely new issue.  

## **Code:**
```           
Graph = Ractive.extend({
    oncomplete: function() {
        this.observe('pts', function(pts) {
            //do some manipulation on pts
            var promise = this.animate('points', _modifiedPts') // promise is undefined
        }
    },
})
```

## Versions affected:
0.8.1 and 0.8.5 
This same code works if I downgrade to 0.7.3

## Platforms affected:
Only tested on Chrome,  sorry don't have other browsers to test with

**Comment by evs-chris** [comment]
I'm having trouble reproducing the issue. Is `_modifiedPts` an object? Can you write up a fiddle or some other example that I can step through?

**Comment by hradhak** [comment]
I've recreated this bug [here](https://jsfiddle.net/tto15hra/10/): 

Sorry, for how much code is in there.  I had a helluva time reproducing this issue without basically recreating my code in the fiddle.   Lines 188-200 is where the animate() is returning undefined instead of a thenable.

**Comment by evs-chris** [comment]
Ah, there was a component mapping involved, and the relevant tests were't covering links. I've added a test to catch that and fixed the issue. This is available now on edge builds (edge, 0.8-dev) and will go out in 0.8.7 when it's ready. Thanks for the report!
